### PR TITLE
improve openapi validation

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/package.scala [hunk 1]
```diff
@@ -29,15 +30,21 @@ package object impl extends SprayJsonSupport with DefaultJsonProtocol {
     UpdateEServiceDescriptorDocumentSeed
   )
 
+  final val entityMarshallerProblem: ToEntityMarshaller[Problem] = sprayJsonMarshaller[Problem]
+
   final val serviceErrorCodePrefix: String = "008"
   final val defaultProblemType: String     = "about:blank"
 
-  def problemOf(httpError: StatusCode, error: ComponentError, defaultMessage: String = "Unknown error"): Problem =
+  def problemOf(
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/package.scala]: most of this function calls use a single error parameter.
I missed where more than one error is passed, but If it is required I suggest to overload `problemOf` to be able to keep the old signature

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/package.scala (hunk 1 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/package.scala]: It's Friday ... I haven't thought about this simple solution.

Thank you!

**Review by galales — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]
LGTM. Isn't this a good moment to maybe move `Problem` and its serde into commons?

**Comment by beetlecrunch** [comment]
> LGTM. Isn't this a good moment to maybe move `Problem` and its serde into commons?

Is it possible? `Problem` depends on code generation and I think is not possible to exclude it from generation.

**Comment by TonioGela** [comment]
> > LGTM. Isn't this a good moment to maybe move `Problem` and its serde into commons?
> 
> Is it possible? `Problem` depends on code generation, and I think is not possible to exclude it from generation.

AFAIK it's just a matter of not using it, in the sense that we can change the import in the templates. TECHNICALLY we won't use the definition we are exposing in the open API spec, but since it's a thing that changes just once a decade, the boilerplate reduction will be worth it. IMHO.

**Pull request merged**